core/src/main/java/org/apache/carbondata/core/datamap/dev/DataMap.java

```diff
@@ -54,6 +56,13 @@ void init(DataMapModel dataMapModel)
   List<T> prune(Expression filter, SegmentProperties segmentProperties,
       List<PartitionSpec> partitions, CarbonTable carbonTable) throws IOException;
 
+  /**
+   * Prune the data maps for finding the row count. It returns a Map of
+   * blockletpath and the row count
+   */
+  Map<String, Integer> pruneRowCount(Segment segment, SegmentProperties segmentProperties,
```

Comment:
It is just getting row count, not pruning anything here so, please rename it accordingly